File: fap_online_backend/fap_online_backend/src/main/java/com/example/demo/dto/ChildDetailDTO.java
package com.example.demo.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChildDetailDTO {
    private Integer studentId;
    private String studentCode;
    private String fullName;
    private String email;
    private String phone;
    private LocalDate dateOfBirth;
    private String gender;
    private String avatarUrl;
    private String major;
    private Integer enrollmentYear;
    private String academicStatus;
}

File: fap_online_ui/lib/models/response/child_detail_dto.dart
class ChildDetailDTO {
  final int studentId;
  final String studentCode;
  final String fullName;
  final String major;
  final String academicStatus;
  final String? email;
  final String? phone;
  final String? dateOfBirth;
  final String? gender;
  final String? address;
  final String? avatarUrl;
  final int? enrollmentYear;

  ChildDetailDTO({
    required this.studentId,
    required this.studentCode,
    required this.fullName,
    required this.major,
    required this.academicStatus,
    this.email,
    this.phone,
    this.dateOfBirth,
    this.gender,
    this.address,
    this.avatarUrl,
    this.enrollmentYear,
  });

  factory ChildDetailDTO.fromJson(Map<String, dynamic> json) {
    return ChildDetailDTO(
      studentId: json['studentId'] ?? 0,
      studentCode: json['studentCode'] ?? '',
      fullName: json['fullName'] ?? '',
      major: json['major'] ?? '',
      academicStatus: json['academicStatus'] ?? '',
      email: json['email'],
      phone: json['phone'],
      dateOfBirth: json['dateOfBirth'],
      gender: json['gender'],
      address: json['address'],
      avatarUrl: json['avatarUrl'],
      enrollmentYear: json['enrollmentYear'],
    );
  }
}
